### PR TITLE
Add event editing form

### PR DESCRIPTION
**Pull request by markspolakovs**
This will require the creation of two permissions, `AUTH_CREATEEVENT` and `AUTH_EDITANYEVENT`. [edit: done on prod/staging]

### CODE REVIEW

**Comment by URY-GitHub** [comment]
Thank you for this pull request, @markspolakovs!

@michael-grace, would you mind reviewing this PR, please? Feel free to ask anyone for help if you're not quite sure what you're doing.

File: src/Classes/ServiceAPI/MyRadio_Event.php [hunk 1]
```diff
@@ -252,4 +380,17 @@ public function toDataSource($mixins = [])
             'master' => (is_null($this->masterId) ? null : self::getInstance($this->masterId)->toDataSource($mixins))
         ];
     }
+
+    public function canWeEdit()
+    {
+        return $this->getHost()->getID() !== MyRadio_User::getCurrentOrSystemUser()->getID()
```
Comment (michael-grace) [inline src/Classes/ServiceAPI/MyRadio_Event.php:386]: Surely this should be EQUALS not NOT EQUAL as if host==current, they can edit

File: src/Controllers/Events/editEvent.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    } else {
+        // edit
+        /** @var MyRadio_Event $event */
+        $event = MyRadio_Event::getInstance($data['id']);
+
+        // check permissions
+        $event->checkEditPermissions();
+
+        if ($data['title'] !== $event->getTitle()) {
+            $event->setTitle($data['title']);
```
Comment (michael-grace) [inline src/Controllers/Events/editEvent.php:30]: Couldn't we just have an "update all", even if it passes some same-as-before info, to take this down to just 1 db hit

**Comment by markspolakovs** [comment]
'Kay, I think all editing buttons work.

**Review by michael-grace — CHANGES_REQUESTED** [review]
Left some commenteoes 🙂

File: src/Controllers/Events/editEvent.php (hunk 2 above)
Comment (markspolakovs) [reply src/Controllers/Events/editEvent.php:30]: Good point, will fix

File: src/Classes/ServiceAPI/MyRadio_Event.php (hunk 1 above)
Comment (markspolakovs) [reply src/Classes/ServiceAPI/MyRadio_Event.php:398]: oops

**Review by michael-grace — APPROVED** [review]

**Pull request merged**